### ISSUE TITLE
Build - enable asan

### ISSUE THREAD

**Issue by rozniak** [rozniak/xfce-winxp-tc#308]
After #306 , it would obviously be a good idea to enable asan for the builds to capture these sorts of issues early.

Will need to add the following to the CMake, probably the common CMakeLists

```CMake
add_compile_options(-fsanitize=address)
add_link_options(-fsanitize=address)
```

and add this for checked/debug builds
```CMake
add_compile_options(-fno-omit-frame-pointer)
```

**Comment by rozniak** [comment]
Think this should be for checked builds only...

**Comment by rozniak** [comment]
Working through some existing issues brought up by ASan

**Comment by rozniak** [comment]
NTS: notepad save as dialog leaks, unsure why - investigate GTK source

**Comment by rozniak** [comment]
NTS: `libfontconfig` leaks so should be suppressed

Useful to test most programs with
`LSAN_OPTIONS=suppressions=supp.txt <the program>`

with `supp.txt` contents

```
leaks:libfontconfig.so
```

When looking for specific leaked mem, report addresses with
`LSAN_OPTIONS=suppressions=supp.txt:report_objects=1 <the program>`

**Comment by rozniak** [comment]
Probably won't have any more time on this for the weekend, will have to continue during the week when I get home.

Few things outstanding:
- Potentially investigate the notepad dialog leaks, though might be an upstream issue
- Look into why WNCK is leaking icon pixmaps, this is very frustrating
- Need to migrate `regsvc` to use `GApplication` so it can accept `--quit` and be debugged better (need graceful exit for LSan to work)

Think most everything else is sorted for now, though this is still a fair bit of work to undertake.

**Comment by rozniak** [comment]
I think I'll save the upstream investigations for a separate issue... I have just migrated `regsvc` to `GApplication` and it seems all good (probably because the `GVariant` leaks were fixed earlier).

Remaining issues for GTK/WNCK leaks can be looked at separately. They might just be 'one off' leaks that are intentional by those libraries.
